### ISSUE TITLE
q-input with date or time mask skips preceding character before deleting on backspace

### ISSUE THREAD

**Issue by kasperskovhansen** [quasarframework/quasar#15594]
### What happened?

When pressing backspace in a q-input with either a date or a time mask, the cursor moves one character to the left before deleting a character.

### What did you expect to happen?

When pressing backspace in a q-input with either a date or a time mask, the character to the left of the cursor should be deleted.

### Reproduction URL

https://stackblitz.com/edit/quasar-starter-2-jszheu?file=src/components/ExampleComponent.vue

### How to reproduce?

1. Go to the provided reproduction link or go to the docs for q-date using q-input https://quasar.dev/vue-components/date#with-qinput for Quasar version 2.11.8 in Chrome
3. Click on the q-input component to focus it
4. Press backspace
5. Observe that the cursor does not delete the preceding character as expected but skips it and deletes the character coming before.

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Components (quasar), Accessibility [a11y] (quasar)

### Platforms/Browsers

Chrome

### Quasar info output

```Shell
Operating System - Linux(5.0 (Macintosh)) - linux/x64
NodeJs - 16.14.2

Global packages
  NPM - 7.17.0
  yarn - 1.22.19
  @quasar/cli - undefined
  @quasar/icongenie - Not installed
  cordova - Not installed

Important local packages
  quasar - 2.11.8 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-vite - 1.0.0-beta.7 -- Quasar Framework App CLI with Vite
  @quasar/extras - 1.13.3 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.31 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.0.14
  pinia - 2.0.12 -- Intuitive, type safe and flexible Store for Vue
  vuex - Not installed
  vite - 2.8.6 -- Native-ESM powered web dev build tool
  eslint - 8.11.0 -- An AST-based pattern checker for JavaScript.
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  *None installed*

Networking
  Host - quasarstarter2jszheu-dw1g
  en0 - 192.168.1.104
```

### Additional context

The bug has been encountered in Quasar version 2.11.8

**Comment by pdanpdan** [comment]
Sorry, too much simplification :)

**Comment by kasperskovhansen** [comment]
Np. Really appreciate what you do🙏

**Comment by pdanpdan** [comment]
The fix is merged already:)

**Comment by KammererTob** [comment]
Thanks for the fix. Is there any estimate when this will also be released for Quasar v1?
